### ISSUE TITLE
AgentLoadException: Agent JAR not found or no Agent-Class attribute

### ISSUE THREAD

**Issue by dzcdhj** [ActiveJpa/activejpa#25]
Caused by: com.sun.tools.attach.AgentLoadException: Agent JAR not found or no Agent-Class attribute
	at sun.tools.attach.HotSpotVirtualMachine.loadAgent(HotSpotVirtualMachine.java:117)
	at org.activejpa.enhancer.ActiveJpaAgentLoader.loadAgent(ActiveJpaAgentLoader.java:40)

windwos xp with jdk 1.7

**Comment by zziahui** [comment]
I want to konw how many users who used it ?

**Comment by ganeshs** [comment]
@zziahui, its been used in production at web scale. I can name a couple of
them, www.flipkart.com, www.hightail.com, www.olacabs.com.


-Ganesh

**Comment by stevexu01** [comment]
Workaround:
 String OS = System.getProperty("os.name");
 String path = codeSource.getLocation().toURI().getPath();
 if(path != null && path.startsWith("/") && OS.startsWith("Windows")){
      path = path.substring(1);
}
vm.loadAgent(path, "");

**Comment by casper-os** [comment]
I got the same issue, until we got the official patch, one way to workaround is create a class org.activejpa.enhancer.ActiveJpaAgentLoaderImpl (override the current implement) - confirmed working on Windows
(and using the code above from @stevexu01 )
`
package org.activejpa.enhancer;

import com.sun.tools.attach.VirtualMachine;

import java.lang.management.ManagementFactory;
import java.security.CodeSource;

public class ActiveJpaAgentLoaderImpl {
    public ActiveJpaAgentLoaderImpl() {
    }

    public static void loadAgent() {
        String nameOfRunningVM = ManagementFactory.getRuntimeMXBean().getName();
        int p = nameOfRunningVM.indexOf(64);
        String pid = nameOfRunningVM.substring(0, p);

        try {
            VirtualMachine vm = VirtualMachine.attach(pid);
            CodeSource codeSource = ActiveJpaAgent.class.getProtectionDomain().getCodeSource();
            String OS = System.getProperty("os.name");
            String path = codeSource.getLocation().toURI().getPath();
            if (path != null && path.startsWith("/") && OS.startsWith("Windows")) {
                path = path.substring(1);
            }
            vm.loadAgent(path, "");
            vm.detach();
        } catch (Exception var5) {
            throw new RuntimeException(var5);
        }
    }
}
`

**Comment by stevenchendy** [comment]
This may not be caused by loading classes, that is, your application running in docker needs to run inside the monitor.
